operating-system.c

```diff
@@ -153,9 +159,16 @@ void write_FRU(struct FRU_DATA *fru, char * file_name, bool packed)
 		if (tmp != len)
 			printf_err ("Didn't write entire file\n");
 	} else {
-		if((fp = fopen(file_name, "wb")) == NULL)
+		if((fp = fopen(file_name, "r+b")) == NULL)
 			printf_err("Cannot open file.\n");
 
+		fseek(fp, 0L, SEEK_END);
+		long size = ftell(fp);
+		rewind(fp);
```

Comment:
Consider using `fstat()`. While this works on posix based systems, this is not "best practices" (and portable way) to get the file size (even more when 'b' is involved).